### PR TITLE
fix: [CDS-80097]: Adding support of fileFilter in AmazonS3 artifact and snow updateMultiple change

### PR DESCRIPTION
**Pull request by satyam-harness**
### Summary
Enahancement - Adding support of fileFilter in amazonS3 artifactPath fetch. Earlier we were supporting fixed val or regex.
Solution -  Added a new field under value section of artifactPath, field is named as fileFilter.

Test cases added - 

- Renders fine for the existing artifact when filePath and fileFilter is present.
- FileFilter field operations - Making fileFilter runtime will make filePath as runtime.
- FileFilter field operations - Making changes to fixed fileFilter will reset filePath

#### Screenshots

https://github.com/harness/harness-core-ui/assets/96036463/b278f604-c176-4c28-9102-31d33968b1db

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger jest

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonS3Artifact/__tests__/AmazonS3.test.tsx [hunk 1]
```diff
@@ -336,7 +437,7 @@ describe('AmazonS3 tests', () => {
     act(() => {
       fireEvent.change(filePathRegexInput, { target: { value: 'test_file_path_regex' } })
     })
-
+    expect(container).toMatchSnapshot()
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonS3Artifact/__tests__/AmazonS3.test.tsx:440]: Can we avoid snapshot testing

**Review by PratyushGar — APPROVED** [review]

**Review by litvinovstas — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks good

**Comment by satyam-harness** [comment]
trigger messagecheck

**Pull request merged**